pandas/core/groupby/generic.py

```diff
@@ -1780,12 +1796,13 @@ def value_counts(
                 indexed_group_size = result_series.groupby(
                     result_series.index.droplevel(levels),
                     sort=self.sort,
-                    observed=self.observed,
                     dropna=self.dropna,
                 ).transform("sum")
-
                 result_series /= indexed_group_size
 
+                # Handle empty groups of non-observed categories
+                result_series = result_series.fillna(0.0)
+
```

Comment:
It fixes an other bug related to `normalize` incorrectly producing NaN on non-observed categories:
```
>>> c1 = pd.Categorical(["a", "b"], categories=["a", "b", "c"])
>>> c2 = range(2)
>>> df = pd.DataFrame({"c1": c1, "c1": c2})
>>> df.groupby("c1").value_counts(normalize=True)
c1  c2
a   0     1.0
    1     0.0
b   1     1.0
    0     0.0
c   0     NaN    # <--- 0 is expected
    1     NaN    # <--- 0 is expected
dtype: float64
```

Reply:
I guess these are by definition NaN here as we reindexed. this seems quite unfortunate as .value_counts() should be integer valued. can we fill on the reindex iteself? (which will keep the int dtype)

Reply:
When normalize is True (this section of code), the values are percentages in [0.0, 0.1]. The nans don't come from a reindex, rather here they are coming from where `indexed_group_size = result_series.groupby(...).transform('sum')` is 0.